### PR TITLE
added forms support

### LINKED ISSUE #149: Posting forms is not working
posting forms is not working

example:

```http
POST http://localhost:8080/upload

-F "uploads=@/home/user/my_file.md"

{}
```
this is translated to the following curl command using RestNvimPreview (which works):
`curl -sSL --compressed -X 'POST' -F "uploads=@/home/user/my_file.md" 'http://localhost:8080/upload'`

but running it using RestNvim returns this error

```lua
Error executing luv callback:
...site/pack/packer/start/plenary.nvim/lua/plenary/curl.lua:260: post http://localhost:8080/upload - curl error exit_code=26 stderr={ "curl: (26) Failed to open/read local data from file/application" }
stack traceback:
	[C]: in function 'error'
	...site/pack/packer/start/plenary.nvim/lua/plenary/curl.lua:260: in function '_user_on_exit'
	.../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:240: in function '_shutdown'
	.../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:46: in function <.../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:37>
```
taking a look at the table that is being sent to plenary's curl.lua I found this

```lua
{
  body = vim.empty_dict(),
  bufnr = 1,
  callback = <function 1>,
  dry_run = false,
  end_line = 16,
  headers = {},
  method = "post",
  raw = { '-F "uploads=@/home/kanon/my_file.md"' },
  start_line = 11,
  url = "http://localhost:8080/upload"
}
```
the option is being sent in the `raw` key, but plenary also has the option of receiving a `form` key, so I'm guessing sending it there would help?

### PR DESCRIPTION
**Pull request by aldevv**
fixes #149

### CODE REVIEW

**Comment by rochacbruno** [comment]
I like this, I also in need of this feature.

This PR is missing a test in the tests/ folder, which also serves as example/doc

**Comment by rochacbruno** [comment]
nevermind @aldevv look #126

**Comment by rochacbruno** [comment]
@teto do you want to have this one merged? I am missing this feature and I can help reviewing once @aldevv fix the conflicts

**Comment by aldevv** [comment]
@teto x2

File: lua/rest-nvim/curl/init.lua [hunk 1]
```diff
@@ -211,6 +211,7 @@ M.curl_cmd = function(opts)
     return
   else
     opts.callback = vim.schedule_wrap(create_callback(opts.method, opts.url))
+
```
Comment (teto) [inline lua/rest-nvim/curl/init.lua:214]: remove

**Comment by teto** [comment]
could you rebase and add tests please ?

**Comment by aldevv** [comment]
hey @teto I rebased and added a test, what do you think?

File: lua/rest-nvim/request/init.lua [hunk 2]
```diff
@@ -66,8 +66,8 @@ local function get_body(bufnr, start_line, stop_line, has_json)
     if line:find("{%%") then
       break
     end
-    -- Ignore commented lines with and without indent
-    if not utils.contains_comments(line) then
+    -- Ignore commented lines with and without indent and Ignore forms
+    if not utils.contains_comments(line) and not line:find("^-F .+") then
```
Comment (teto) [inline lua/rest-nvim/request/init.lua:70]: we may not want to add those to body but we surely want to replace variables, dont we ?

File: lua/rest-nvim/request/init.lua [hunk 3]
```diff
@@ -21,7 +21,7 @@ local function get_importfile_name(bufnr, start_line, stop_line)
     local fileimport_spliced
     fileimport_line = vim.api.nvim_buf_get_lines(bufnr, import_line - 1, import_line, false)
     fileimport_string =
-    string.gsub(fileimport_line[1], "<", "", 1):gsub("^%s+", ""):gsub("%s+$", "")
+      string.gsub(fileimport_line[1], "<", "", 1):gsub("^%s+", ""):gsub("%s+$", "")
```
Comment (teto) [inline lua/rest-nvim/request/init.lua:24]: useless change

Comment (aldevv) [reply lua/rest-nvim/request/init.lua:24]: got this running stylua following stylua.toml

**Comment by davidsierradz** [comment]
Hi! Would be nice to also support `application/x-www-form-urlencoded` forms with the `-d` Curl flag

**Comment by boltlessengineer** [comment]
Closing this due to v3 release for both rest.nvim and tree-sitter-http.
Now `tree-sitter-http` supports both `form-urlencoded` and `multipart-form-data` following intellij http client’s spec

**Pull request closed without merging**